haystack/marshal/yaml.py

```diff
@@ -14,14 +14,33 @@ def construct_python_tuple(self, node: yaml.SequenceNode):
         return tuple(self.construct_sequence(node))
 
 
+class YamlDumper(yaml.SafeDumper):  # pylint: disable=too-many-ancestors
+    def represent_tuple(self, data: tuple):
+        """Represent a Python tuple."""
+        return self.represent_sequence("tag:yaml.org,2002:python/tuple", data)
+
+
+YamlDumper.add_representer(tuple, YamlDumper.represent_tuple)
```

Comment:
Does this actually work? I'm getting the same output locally whether I use this dumper or not. 🤔 

```
In [5]: yaml.dump((1,2,3,4), Dumper=YamlDumper)
Out[5]: '!!python/tuple\n- 1\n- 2\n- 3\n- 4\n'

In [6]: yaml.dump((1,2,3,4))
Out[6]: '!!python/tuple\n- 1\n- 2\n- 3\n- 4\n'
```

Reply:
The default dumper is the non-safe variant which serializes tuples with that particular tag. The safe dumper (which rejects non-basic Python objects) coverts tuples to lists, which breaks existing pipelines. We avoid this by subclassing the default safe dumper and overriding how the tuples are represented.

That call inside `represent_tuple`  is directly pulled from the non-safe dumper's code, which is why the outputs are same (it's intentional).